modules/h2/src/main/java/org/apache/ignite/internal/processors/query/h2/H2MemoryTracker.java

```diff
@@ -16,21 +16,68 @@
 
 package org.apache.ignite.internal.processors.query.h2;
 
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+
 /**
- * Abstract memory tracker.
+ * Memory tracker.
  */
 public abstract class H2MemoryTracker implements AutoCloseable {
+    /** Objects to be closed along with the current tracker. */
+    private Collection<Closeable> closeList;
```

Comment:
Won't `AutoCloseable` be enough here instead of `Closeable`?

Reply:
Agree.